amazon-kinesis-client/src/test/java/software/amazon/kinesis/utils/LeaseTableManager.java

```diff
@@ -0,0 +1,74 @@
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableRequest;
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableResponse;
+import software.amazon.awssdk.services.dynamodb.model.ListTablesRequest;
+import software.amazon.awssdk.services.dynamodb.model.ListTablesResponse;
+import software.amazon.awssdk.services.dynamodb.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.dynamodb.model.TableStatus;
+import software.amazon.kinesis.common.FutureUtils;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class LeaseTableManager extends AWSResourceManager {
+
+    private final DynamoDbAsyncClient dynamoClient;
+
+    public LeaseTableManager(DynamoDbAsyncClient dynamoClient) throws URISyntaxException, IOException {
+        this.dynamoClient = dynamoClient;
+    }
+
+    public boolean _isResourceActive(String tableName) {
+        final DescribeTableRequest request = DescribeTableRequest.builder().tableName(tableName).build();
+        final CompletableFuture<DescribeTableResponse> describeTableResponseCompletableFuture = dynamoClient.describeTable(request);
+
+        try {
+            final DescribeTableResponse response = describeTableResponseCompletableFuture.get(30, TimeUnit.SECONDS);
+            boolean isActive = response.table().tableStatus().equals(TableStatus.ACTIVE);
+            if (!isActive) {
+                throw new RuntimeException("Table is not active, instead in status: " + response.table().tableStatus());
+            }
+            return true;
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof ResourceNotFoundException) {
+                return false;
+            } else {
+                throw new RuntimeException(e);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void _deleteResource(String tableName) throws Exception {
+        final DeleteTableRequest request = DeleteTableRequest.builder().tableName(tableName).build();
+        FutureUtils.resolveOrCancelFuture(dynamoClient.deleteTable(request), Duration.ofSeconds(60));
+    }
+
+    public List<String> _getAllResourceNames() throws Exception {
+        List<String> tableNames = new ArrayList<>();
+        final ListTablesRequest request = ListTablesRequest.builder().build();
+        ListTablesResponse response = null;
+        while(response == null || response.lastEvaluatedTableName() != null) {
+            try {
+                response = FutureUtils.resolveOrCancelFuture(dynamoClient.listTables(request), Duration.ofSeconds(60));
+            } catch (ExecutionException | InterruptedException e) {
```

Comment:
Fixing the request that is made. With this new change there are many places in this method that are resource dependent (the initialization at the top, while condition, try block, addAll call...) that would need to be implemented in the resource specific managers that I think that abstracting this would harm readability more than it would help.